staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -91,6 +91,20 @@ func IsExternal(name string) bool {
 	return name == externalCloudProvider
 }
 
+// IsDeprecatedInternal is responsible for preventing cloud.Interface
+// from being initialized in kubelet, kube-controller-manager or kube-api-server
+func IsDeprecatedInternal(name string) bool {
+	for _, provider := range deprecatedCloudProviders {
+		if provider.name == name {
+			detail := fmt.Sprintf("Please use 'external' cloud provider for %q: %s", name, provider.detail)
+			klog.Warningf("WARNING: %q built-in cloud provider is now disabled. %s", name, detail)
+			return true
```

Comment:
Currently we are ignoring the value of provider.external. That means that any alpha tests for a provider with external=false are expected to fail to have the cluster come up successfully. OTOH not sure the feature flag meets its use case if we do anything else.

Reply:
The alpha tests will have to set and unset the FF independently from each other for every cloud provider testing. Having to rely on internal switch is useful when we are not confident enough out-of-tree implementation will fully substitute existing in-tree for the provider. I don't think this is the case for any of the cloud providers, but if it will be the case we could add `provider.external` dependency later before moving feature gate out of alpha stage.